### ISSUE TITLE
Sonarqube format incorrect column index

### ISSUE THREAD

**Issue by sushant-pradhan** [hadolint/hadolint#658]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
The [newly introduced](https://github.com/hadolint/hadolint/pull/641) `sonarqube` format to push issues to Sonarqube as a [generic issue ](https://docs.sonarqube.org/latest/analysis/generic-issue/)is super useful. However, there is a need to change column index which is currently 1 based to the expected 0 based.

Related documentations/issues:
https://docs.sonarqube.org/latest/analysis/generic-issue/
https://community.sonarsource.com/t/error-while-import-external-issues-without-endline-and-endcolumn/9007

### Actual behavior
For the Dockerfile:
```docker
# Step 1: pull from dockerhub
# Step 2: copy the already downloaded packages
FROM python:3.7
ADD dist/pip-packages /tmp/pip-packages
```
Run the following command to generate report in sonarqube format:
`hadolint -f sonarqube Dockerfile > hadolint-issues.json`

Contents of `hadolint-issues.json`
```json
{
   "issues":[
      {
         "primaryLocation":{
            "message":"Use COPY instead of ADD for files and folders",
            "filePath":"Dockerfile",
            "textRange":{
               "endColumn":1,
               "startLine":4,
               "endLine":4,
               "startColumn":1
            }
         },
         "ruleId":"DL3020",
         "severity":"CRITICAL",
         "engineId":"Hadolint",
         "type":"BUG"
      }
   ]
}
```
Note that the startColumn is 1 (it should be 0) as per the Sonar documentation:

![image](https://user-images.githubusercontent.com/16751240/122690990-6c0cc880-d1f2-11eb-9336-c7f0f4a843d9.png)

### Steps to reproduce the behavior

Dockerfile contents:
```docker
# Step 1: pull from dockerhub
# Step 2: copy the already downloaded packages
FROM python:3.7
ADD dist/pip-packages /tmp/pip-packages
```
Run hadolint on the Dockerfile:
`hadolint -f sonarqube Dockerfile > hadolint-issues.json`
Push the scan results to Sonarqube:
`sonar-scanner -Dsonar.externalIssuesReportPaths=hadolint-issues.json <other ags redacted>`
Sonar fails with error:
```bash
..
ERROR: Error during SonarScanner execution
      java.lang.IllegalArgumentException: Start pointer [line=4, lineOffset=1] should be before end pointer [line=4, lineOffset=1]
...
```
On changing the output format to 0 based  (setting startColumn = 0), the scan passes successfully:
Modified hadolint-issues.jsob (to pass sonar):
```json
{
   "issues":[
      {
         "primaryLocation":{
            "message":"Use COPY instead of ADD for files and folders",
            "filePath":"Dockerfile",
            "textRange":{
               "endColumn":1,
               "startLine":4,
               "endLine":4,
               "startColumn":0
            }
         },
         "ruleId":"DL3020",
         "severity":"CRITICAL",
         "engineId":"Hadolint",
         "type":"BUG"
      }
   ]
}
```
Sonar version: 7.9.1
Sonar cli: 4.6.2.2472
Hadolint: v2.5.0

**Comment by m-ildefons** [comment]
Hi @sushant-pradhan,
thanks for the detailed bugreport. I've submitted a PR with a fix and the next release should not suffer from this issue anymore.
As an aside, the start column does not *have* to be 0, it just has to be truly less than the end column, unfortunately that was not documented in the spec.
